coordinator/src/services/translation/request/payload/factory.ts

```diff
@@ -0,0 +1,203 @@
+    return getMessageIdFromBundle(bundle)
+  }
+
+  create(bundle: fhir.Bundle, logger: pino.Logger): Payload<BundleTranslationResult> {
+    const messageType = identifyMessageType(bundle)
+
+    switch (messageType) {
+      case fhir.EventCodingCode.PRESCRIPTION:
+        return {
+          content: this.createParentPrescription(bundle, logger),
+          interactionId: hl7V3.Hl7InteractionIdentifier.PARENT_PRESCRIPTION_URGENT
+        }
+
+      case fhir.EventCodingCode.CANCELLATION:
+        return {
+          // TODO: pass logger?
+          content: this.createCancellation(bundle),
+          interactionId: hl7V3.Hl7InteractionIdentifier.CANCEL_REQUEST
+        }
+
+      case fhir.EventCodingCode.DISPENSE:
+        return {
+          content: this.createDispenseNotification(bundle, logger),
+          interactionId: hl7V3.Hl7InteractionIdentifier.DISPENSE_NOTIFICATION
+        }
+    }
+  }
+
+  private createParentPrescription(bundle: fhir.Bundle, logger: pino.Logger): hl7V3.ParentPrescriptionRoot {
+    const parentPrescription = convertParentPrescription(bundle, logger)
+    return new hl7V3.ParentPrescriptionRoot(parentPrescription)
+  }
+
+  private createCancellation(bundle: fhir.Bundle): hl7V3.CancellationRequestRoot {
+    const cancellationRequest = convertCancellation(bundle)
+    return new hl7V3.CancellationRequestRoot(cancellationRequest)
+  }
+
+  private createDispenseNotification(bundle: fhir.Bundle, logger: pino.Logger): hl7V3.DispenseNotificationRoot {
+    const dispenseNotification = convertDispenseNotification(bundle, logger)
+    return new hl7V3.DispenseNotificationRoot(dispenseNotification)
+  }
+}
+
+class TaskTranslationResultFactory extends PayloadFactory {
+  getPayloadId(task: fhir.Task): string {
+    return getMessageIdFromTask(task)
+  }
+
+  create(task: fhir.Task): Payload<TaskTranslationResult> {
+    switch (task.status) {
+      case fhir.TaskStatus.REJECTED:
+        return {
+          content: this.createDispenseProposalReturn(task),
+          interactionId: hl7V3.Hl7InteractionIdentifier.DISPENSE_PROPOSAL_RETURN
+        }
+      case fhir.TaskStatus.IN_PROGRESS:
+        return {
+          content: this.createDispenserWithdraw(task),
+          interactionId: hl7V3.Hl7InteractionIdentifier.DISPENSER_WITHDRAW
+        }
+    }
+  }
+
+  private createDispenseProposalReturn(task: fhir.Task): hl7V3.DispenseProposalReturnRoot {
+    const dispenseProposalReturn = convertTaskToDispenseProposalReturn(task)
+    return new hl7V3.DispenseProposalReturnRoot(dispenseProposalReturn)
+  }
+
+  private createDispenserWithdraw(task: fhir.Task): hl7V3.EtpWithdrawRoot {
+    const etpWithdraw = convertTaskToEtpWithdraw(task)
+    return new hl7V3.EtpWithdrawRoot(etpWithdraw)
+  }
+}
+
+class ParametersTranslationResultFactory extends PayloadFactory {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  getPayloadId(parameters: fhir.Parameters): string {
```

Comment:
im not sure typescript will complain if you don't provide all parameters? (could be wrong)

Reply:
Added that because of linter warnings